src/experimental/information_estimates.cpp

```diff
@@ -0,0 +1,568 @@
+                    low_pt_x = low_pt_x < 0.0 ? 0.0 : low_pt_x;
+                    // Maximum X point
+                    kt_double max_pt_x = robot_pose_raw.GetX() + m_max_sensor_range; // X: Posicion del scan + Rango maximo (Para saber si el scan ve por fuera del grid)
+                    max_pt_x = max_pt_x > dist_x ? dist_x : max_pt_x;
+                    // Lowest Y point
+                    kt_double low_pt_y = robot_pose_raw.GetY() - m_max_sensor_range; // Y: Posicion del scan - Rango maximo (Para saber si el scan ve por fuera del grid)
+                    low_pt_y = low_pt_y < 0.0 ? 0.0 : low_pt_y;
+                    // Maximum Y point
+                    kt_double max_pt_y = robot_pose_raw.GetY() + m_max_sensor_range; // Y: Posicion del scan + Rango maximo (Para saber si el scan ve por fuera del grid)
+                    max_pt_y = max_pt_y > dist_y ? dist_y : max_pt_y;
+
+                    karto::Vector2<kt_double> min_pt{low_pt_x, low_pt_y}; // Esto es distancia no posicion de celda
+                    karto::Vector2<kt_double> max_pt{max_pt_x, max_pt_y}; // Esto es distancia no posicion de celda
+
+                    // Esto podia ser un karto::Pose2
+                    // karto::Vector2<int> getGridPosition(karto::Vector2<kt_double> const& pose, kt_double resolution)
+                    karto::Vector2<int> min_cell = utils::grid_operations::getGridPosition(min_pt, m_cell_resol);
+                    karto::Vector2<int> max_cell = utils::grid_operations::getGridPosition(max_pt, m_cell_resol);
+
+                    // i and j are the current row and column
+                    if ((i < min_cell.GetX() || i > max_cell.GetX()) || (j < min_cell.GetY() || j > max_cell.GetY()))
+                    {
+                        continue;
+                    }
+
+                    // Current scan laser readings
+                    karto::PointVectorDouble laser_readings = scan->GetPointReadings(true);
+
+                    karto::Pose2 robot_pose(robot_pose_raw.GetX() - m_low_x, robot_pose_raw.GetY() - m_low_y, robot_pose_raw.GetHeading());
+                    karto::Vector2<int> robot_grid_position = utils::grid_operations::getGridPosition(robot_pose.GetPosition(), m_cell_resol);
+
+                    /*
+                        Ahora que estoy dentro de un grid local, debo hacer una transformacion simple de
+                        la posicion del robot y de cada uno de los lasers, para asi poder compararlos correctamente
+                        con cada una de las celdas.
+                    */
+
+                    // ----- Point 3 -----
+                    // I could use some kind of angle to filter the region where the scan is.
+                    // I can use the range based for here
+                    for (int i = 0; i < laser_readings.size(); ++i)
```

Comment:
@kmilo7204 meta: hmm, I don't follow this bit. If you know where the cell is, where the scan was acquired, and how it is laid out in memory, why not simply computing the index of the ray(s) that need to be looked at? 

You can do that in the grid frame or in the scan frame. Perhaps the latter is easiest, you can then compute the angles of cell corners and use rays that fall in the same angular interval. Does that make sense?